### DISCUSSION TITLE
IRON MAN MODE

### DISCUSSION THREAD

**Discussion by Alkanty** [FakeFishGames/Barotrauma#13031] [Ideas and suggestions]
The game is great in all its way but i feel (personnaly) it miss one thing... An optional iron Man mode like FTL.

**Mechanics:** 

- If all the crew member dies = Gameover and save game is erased
-  if you ALT F4 or quit before it erase the save game 
- you can only save in outpost or different conditions 
- if escape shuttle reach the station = saved you get your sub back (minus a cost maybe ? or buy a new one if you have none in your garage  ) this give a new purpose of having a sub with a shuttle in it and it gives a new purpose for having multiple ship. 
- keep same death system as it would work well with it. 
- Able to sell your prior ship so you can make more money from it ! 
- special steam achievement for Iron man mode
- Count of how far you went at the game over screen. 

**Extrapolation**

- special respawn pod event for rescue mission in an attempt to bring back your sub if all of your party died. can respawn via a rescue pod that drop player near submarine and exit. Cost money (act a bit like a "get out of jail" card)

**Justification**
- make more replayability for more hardcore gamers
- gives more weight to descision making as a crew in single and multiplayer (should i do this mission ? is it too risky ? )
- gives more depth in buying submarine (choice that has consequence) (should i sell my old sub or keep them in case of an emergency ?)
- More responsibility for the captain in multiplayer ! 

 _this would make game realy hard but it could make it fun too  (for all the challenge lovers out there) and gives a more strategic approach to the game. In barotrauma there is "trauma" in it after all ;) 

Thank you for reading

PS : sorry for my bad english as i am french :3

**Comment by iridescent-lightning** [comment]
Exactly! I will probably look into the source code to see if I can tackle them one or two.

**Reply by Alkanty** [reply]
are you a modder or member of the dev team ? I wonder if the respawn options are hard coded or not into the game.

**Reply by iridescent-lightning** [reply]
I'm a modder but I do C# modding. Check out  Barotrauma Die Hard.

**Reply by Alkanty** [reply]
wow pretty neat indeed i play a lot of modded baro and never came across it. i like a lot of your ideas, might test it with friend on a new campaign. We are playing wih _M_ mod at the moment. If you see something that can be done with the save and selling ship system tell me ! 
- for the selling : I guess it should be possible as i know you can delete bought submarines from savefile. Now to know how to synchronise between a in game shop and the fact that it trade money out for delete the sub in the file
- for the iron mod i feel it can get a bit more complicated without dev intervention as the server options seems hard coded (but hey modder makers miracle) 

anyway great talent to be able to master C#, lua , and xml, i would love too if only i was a bit smarter (and more patient xD)

**Reply by iridescent-lightning** [reply]
Currently, the mod is very unbalanced. Because I don't have time other than writing cs. The only sub that has an air recycle system is the die hard version of Barsuk. Going on other subs you will get poisoned by CO2. If you have broken batteries, you can get poisoned by Chlorine as well.  Although opening the door can buy you some time. Because they do spread from hull to hull.

the sonar needs to be balanced as well. Currently it's too narrow. Omni sonar bounces all over your sub and your monitor has lots of blips.

But lots of features only exist in this mod.

I can name a few so you can get some quick preview:
-lamps, junctionboxes are shootable
-low health lamps flicker
-reactor, pump, oxygengenerator needs addtional parts to work.
-you can use key to lock/unlock cabs
-many items have body(divingsuit, diehard version tables and other furniture decor...) you can trip over if you run to them. some need to be in ragdoll mode to interact. Otherwise you can't move in a 2D room.
-gunshot extremely deadly
-armor/healmet loses condition when hit. If bullet doesn't penetrate it, almost no damage is dealt.
-submarine design and payload matters(check it using captain manual)
-vertical engine.(search vertical booster or vertical thruster) the vertical engine was re-worked so the lever on die hard version sub doesn't work. the output signal is wrong.
-you can buy real upgrades from shop. there are slots in sub so you can buy new device and install them. same for vertical engine.
-leaking logic rewokr. now only up to 60% damage your sub starts to leak.
-no sea bed. going down means death (why am I changing the leaking?)
-breakable turrets. hammerheads love turrets or hate them. Or could be both.
-many sounds effect, (opening containers, dropping items)
...
I will add if I can recall.
And yes, this mod runs smoothly. Smoother than many big mods because many core logics were written directly in Cs.
Expected to encounter some crash or bug.

**Reply by Alkanty** [reply]
well i will def look at it !
